### PR TITLE
Rewrite Tick formatters example

### PR DESCRIPTION
**Pull request by timhoffm**
- Collect all relevant descriptions in the summary text at the top
- Merge everything into one figure. The code should be mostly irrelevant for the example. The figure can serve as a standalone reference.

<s>Also closes 25967 by deleting the "Dollar ticks" example.</s> [Deferred]

### CODE REVIEW

File: galleries/examples/ticks/tick-formatters.py [hunk 1]
```diff
@@ -8,6 +8,23 @@
 
 This example illustrates the usage and effect of the most common formatters.
 
+The tick format is configured via the function `~.Axis.set_major_formatter`
+or `~.Axis.set_minor_formatter`. It accepts:
+
+- a format string (this implicitly creates a `.StrMethodFormatter`)
```
[the remark this answers is not in the document]
Comment (timhoffm) [reply galleries/examples/ticks/tick-formatters.py]: Done.

**Comment by timhoffm** [comment]
> Since it's all getting combined and the code doesn't matter, is there anyway to loop over the arguments so that the titles have to stay in sync w/ the code?

I don't think so. While https://matplotlib.org/stable/gallery/ticks/date_formatters_locators.html does something like that, the special code for FuncFormatter and FixedFormatter make it difficult here.

I'm not at all worried about "stay in sync". This will hardly change and the title and formatter code are always in neighboring lines.

**Comment by timhoffm** [comment]
New example: https://output.circle-artifacts.com/output/job/7e5ea6a8-7123-4505-9742-353df3e7644a/artifacts/0/doc/build/html/gallery/ticks/tick-formatters.html#sphx-glr-gallery-ticks-tick-formatters-py

**Comment by timhoffm** [comment]
AFAIK this is not possible in sphinx-gallery. What I would like even more would be an option to collapse code blocks in sphinx-gallery, but that's also not supported.

**Comment by rcomer** [comment]
According to this page, if there is only one code block the image appears before the code.
https://sphinx-gallery.github.io/stable/auto_examples/plot_1_exp.html#sphx-glr-auto-examples-plot-1-exp-py

I am not clear how it is counting more than one code block for the current case 😕

**Comment by timhoffm** [comment]
Likely it counts:
- Title, description
- Code
Which works here: https://matplotlib.org/devdocs/gallery/text_labels_and_annotations/fancytextbox_demo.html#sphx-glr-gallery-text-labels-and-annotations-fancytextbox-demo-py

In this example we have the References text block after the code. Likely that prevents the reordering, which makes sense because it would bring two description blocks together that we have written separately.

**Comment by rcomer** [comment]
I think the reference section is no longer needed to generate the mini-galleries.  For example [plt.rcdefaults](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.rcdefaults.html#matplotlib.pyplot.rcdefaults)'s gallery shows two examples that do not have the explicit reference list.  These days the functions within the code work as clickable links, so maybe it gets it from that?

**Comment by timhoffm** [comment]
The references section is gone.

I've additionally removed the "Dollar ticks" example (closes #25967). The formatting is more concisely explained in this overview. We don't need individual examples for all the formatters.

**Comment by rcomer** [comment]
The dollar ticks example is included here:

https://github.com/matplotlib/matplotlib/blob/5f297631c2f295b2f3b52cfddeb33f02567a07f5/galleries/tutorials/artists.py#L719-L725

**Comment by timhoffm** [comment]
I've restored the dollar ticks example. Let's handle this in another PR.

File: galleries/examples/ticks/tick-formatters.py [hunk 2]
```diff
@@ -8,6 +8,23 @@
...
+  - `.NullFormatter`: No labels on the ticks.
+  - `.StrMethodFormatter`: Use string `str.format` method.
+  - `.FormatStrFormatter`: Use %-style formatting.
+  - `.FuncFormatter`: Define labels through a function.
+  - `.FixedFormatter`: Set the label strings explicitly.
+  - `.ScalarFormatter`: Default formatter for scalars: auto-pick the format string.
+  - `.PercentFormatter`: Format labels as a percentage.
+
+  See :ref:`formatters` for a complete list.
+
```
[the remark this answers is not in the document]
Comment (timhoffm) [reply galleries/examples/ticks/tick-formatters.py:27]: The list is a list of formatters. So indentation is correct IMHO

**Pull request merged**